### DISCUSSION TITLE
No Ping/CoAP response from some devices

### DISCUSSION THREAD

**Discussion by FHFS** [openthread/openthread#10031] [Q&A] [answered]
Hi everyone,

I am trying to use my OpenThread device(nRF52840 and nRF7002DK) to send a GET Request to a CoAP server on the local network.

When I connect to a CoAP server running on my laptop everything works ok. I can get ping responses and the payload from my CoAP GET request. 
However when I try to Ping or CoAP GET the CoAP server on my ESP32 I cannot get any responses. The CoAP GET request arrives on the server.
The ESP32 is connected to the LAN through the same wifi as the laptop. The CoAP server on the ESP32 is working fine from devices connected through the same wifi.

The problem only occurs when a device on the OpenThread network tries to connect to the ESP32.

I am hoping to find an answer or solution here. Any help or suggestion is appreciated!

Also see my discussions on [Devzone](https://devzone.nordicsemi.com/f/nordic-q-a/109931/coap-response-on-openthread-network/) and [esp-matter](https://github.com/espressif/esp-matter/issues/901)

**Comment by jwhui** [comment]
Are you able to provide a packet capture?

Do you see a CoAP response in the packet capture?

**Reply by FHFS** [reply]
Maybe if I turn my laptop into a hotspot I can get a packet capture of the OpenThread Device to the ESP32.
I will get back to you.

Or should I use a OpenThread sniffer?

**Reply by FHFS** [reply]
[captureCoAPGET.zip](https://github.com/openthread/openthread/files/15013731/captureCoAPGET.zip)
Was the sniffer set up ok?

I captured this with a nrf52840 stick with ot-rcp. Google Nesthub as OTBR. A nrf52840 on the OpenThread network sending a CoAP GET to my ESP32 on the LAN through wifi.

**Reply by FHFS** [reply]
[CaptureCoAP.zip](https://github.com/openthread/openthread/files/15063184/CaptureCoAP.zip)
@jwhui I got my router to forward capture to wireshark. Here I captured the CoAP GET request from the OpenThread device to the CoAP server on the ESP32

[captureIPv6Working.zip](https://github.com/openthread/openthread/files/15063462/captureIPv6Working.zip)
This is the capture of the OpenThread device sending a CoAP get to my laptop (which works great)

**Reply by FHFS** [reply, marked as answer]
The ESP32 did not have route hook enabled [see esp-matter issue](https://github.com/espressif/esp-matter/issues/901)
There was no route back to the OpenThread device. With the route hook the ESP32 listens to the RA and now there is a route back to the OpenThread device.
